inline-test-project/src/main/java/org/example/MemoryReadTest.java

```diff
@@ -58,7 +60,16 @@ public static void main(String[] args) throws Exception {
                 totalBytesRead += bytesRead;
                 buffer.clear();
             }
-            System.out.println("Downlaoded(" + i + "): " + totalBytesRead);
+            System.out.println("Downloaded(" + i + "): " + totalBytesRead);
+            if (i % 5000 == 0) {
+                times5000ObjectsDownloaded++;
+                System.out.println("Downloaded 5000 objects. Free memory before garbage collection is: "
```

Comment:
Slight change, it would be helpful to print the other metrics as well:

```
 private void printMemory() {
    int mb = 1024 * 1024;
    long totalMemory = Runtime.getRuntime().totalMemory() / mb;
    long freeMemory = Runtime.getRuntime().freeMemory() / mb;
    long maxMemory = Runtime.getRuntime().maxMemory() / mb;
    MemoryMXBean memoryMXBean = ManagementFactory.getMemoryMXBean();
    MemoryUsage heapMemoryUsage = memoryMXBean.getHeapMemoryUsage();
    long hmemUsage = heapMemoryUsage.getUsed() / (1024 * 1024);
    long heapCommitted = heapMemoryUsage.getCommitted() / (1024 * 1024);
    System.out.println(
        String.format(
            "TotalMemory=%sMB; FreeMemory=%sMB; MaxMemory=%sMB; heamUsage=%s; heapCommitted=%s",
            totalMemory, freeMemory, maxMemory, hmemUsage, heapCommitted));
  }
```